src/ConnectionRequest.jl

```diff
@@ -8,6 +8,39 @@ using ..ConnectionPool
 using MbedTLS: SSLContext
 import ..@debug, ..DEBUG_LEVEL
 
+# hasdotsuffix reports whether s ends in "."+suffix.
+hasdotsuffix(s, suffix) = length(s) > length(suffix) && s[length(s)-length(suffix)] == '.' && s[length(s)-length(suffix)+1:end] == suffix
+
+function isnoproxy(host)
+    for x in NO_PROXY
+        hasdotsuffix(host, x) && return true
+    end
+    return false
+end
+
+function __init__()
+    # check for no_proxy environment variable
+    if haskey(ENV, "no_proxy")
+        for x in split(ENV["no_proxy"], ","; keepempty=false)
+            push!(NO_PROXY, startswith(x, ".") ? x[2:end] : x)
+        end
+    end
+    return
+end
+
+function getproxy(scheme, host)
+    if scheme == "http" && haskey(ENV, "http_proxy")
```

Comment:
```suggestion
    isnoproxy(host) && return nothing
    if scheme == "http" && haskey(ENV, "http_proxy")
```

And remove the `if`s below.